internal/livelint/run_checks.go

```diff
@@ -35,12 +35,18 @@ func (n *Livelint) RunChecks(namespace, deploymentName string, isVerbose bool) e
 		return fmt.Errorf("error getting Deployment Pods: %w", err)
 	}
 
-	// Are you hitting the ResourceQuota limits?
-	result := n.checkAreResourceQuotasHit(namespace, deploymentName)
+	// Is the number of running pods correct ?
+	result := n.CheckIsNumberOfPodsMatching(namespace, deploymentName)
 	statusMsg.AddCheckResult(result)
 	n.ui.Send(statusMsg)
 	if result.HasFailed {
-		return nil
+		// Are you hitting the ResourceQuota limits?
+		result = n.checkAreResourceQuotasHit(namespace, deploymentName)
+		statusMsg.AddCheckResult(result)
+		n.ui.Send(statusMsg)
+		if result.HasFailed {
```

Comment:
Result has been overwritten by the call to `checkAreResourceQuotasHit(...)` shouldn't this path exit in any case, no matter if the quota check failed or not? Or should the other checks always be run if the reason for having a different number of pods isn't the correct one?

I believe this might be an issue:
- Too many pods, quota not exceeded => will continue despite the error - or is this not a leaf condition?

Reply:
we still want to continue debugging to find the issue